Imagr/MainController.py

```diff
@@ -1179,6 +1330,31 @@ def downloadAndInstallPackage(self, url, target, progress_method=None, additiona
             # Clean up after ourselves
             shutil.rmtree(temp_dir)
 
+
+    def downloadDMG(self, url, target):
+        if not os.path.basename(url).endswith('.dmg'):
+            self.errorMessage = "%s doesn't end with either '.dmg'" % url
+            return False
+        if os.path.basename(url).endswith('.dmg'):
```

Comment:
This test is unnecessary: you already returned False if the os.path.basename(url).endswith('.dmg') doesn't end with '.dmg'

Reply:
rewrote